Apps/phvisualping/visualping_connector.py

```diff
@@ -0,0 +1,428 @@
+        # There is a high chance of a PROXY in between phantom and the rest of
+        # world, in case of errors, PROXY's return HTML, this function parses
+        # the error and adds it to the action_result.
+        if 'html' in r.headers.get('Content-Type', ''):
+            return self._process_html_response(r, action_result)
+
+        # it's not content-type that is to be parsed, handle an empty response
+        if not r.text:
+            return self._process_empty_response(r, action_result)
+
+        # everything else is actually an error at this point
+        message = "Can't process response from server. Status Code: {0} Data from server: {1}".format(
+            r.status_code,
+            r.text.replace('{', '{{').replace('}', '}}')
+        )
+
+        return RetVal(action_result.set_status(phantom.APP_ERROR, message), None)
+
+    def _make_rest_call(self, endpoint, action_result, method="get", **kwargs):
+        # **kwargs can be any additional parameters that requests.request accepts
+
+        config = self.get_config()
+
+        resp_json = None
+
+        try:
+            request_func = getattr(requests, method)
+        except AttributeError:
+            return RetVal(
+                action_result.set_status(phantom.APP_ERROR, "Invalid method: {0}".format(method)),
+                resp_json
+            )
+
+        # Create a URL to connect to
+        url = "{}{}".format(self._base_url, endpoint)
+
+        try:
+            r = request_func(
+                url,
+                auth=self._auth,  # basic authentication
+                verify=config.get('verify_server_cert', False),
+                **kwargs
+            )
+        except requests.exceptions.InvalidSchema:
+            error_message = 'Error connecting to server. No connection adapters were found for %s' % (url)
+            return RetVal(action_result.set_status(phantom.APP_ERROR, error_message), resp_json)
+        except requests.exceptions.InvalidURL:
+            error_message = 'Error connecting to server. Invalid URL %s' % (url)
+            return RetVal(action_result.set_status(phantom.APP_ERROR, error_message), resp_json)
+        except requests.exceptions.ConnectionError:
+            error_message = 'Error Details: Connection refused from the server'
+            return RetVal(action_result.set_status(phantom.APP_ERROR, error_message), resp_json)
+        except Exception as e:
+            error_message = self._get_error_message_from_exception(e)
+            return RetVal(
+                action_result.set_status(
+                    phantom.APP_ERROR, "Error Connecting to server. {0}".format(error_message)
+                ), resp_json
+            )
+
+        return self._process_response(r, action_result)
+
+    def _handle_test_connectivity(self, param):
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        self.save_progress("Connecting to endpoint")
```

Comment:
You can add which endpoint you are specifically trying to connect to

Reply:
Fixed